src/frontend/src/utils/scan_range.rs

```diff
@@ -58,6 +61,45 @@ impl ScanRange {
             range: full_range(),
         }
     }
+
+    pub fn try_compute_vnode(
+        &self,
+        dist_key_indices: &[usize],
+        pk_indices: &[usize],
+    ) -> Option<VirtualNode> {
+        if dist_key_indices.is_empty() {
+            return None;
+        }
+
+        let dist_key_in_pk_indices = dist_key_indices
+            .iter()
+            .map(|&di| {
+                pk_indices
+                    .iter()
+                    .position(|&pi| di == pi)
+                    .unwrap_or_else(|| {
+                        panic!(
+                            "distribution keys {:?} must be a subset of primary keys {:?}",
+                            dist_key_indices, pk_indices
+                        )
+                    })
+            })
+            .collect_vec();
+        let pk_prefix_len = self.eq_conds.len();
+        if dist_key_in_pk_indices.iter().any(|&i| i >= pk_prefix_len) {
+            return None;
+        }
```

Comment:
Could this happen?

Reply:
The length of `pk_prefix` in `scan_range` can be `0~pk_len`, which can be shorter than dist key